src/core/tests/Test_TelemetryWriter.py

```diff
@@ -101,21 +103,15 @@ def test_write_event_msg_size_limit(self):
             self.assertTrue("a"*(len(message.encode('utf-8')) - chars_dropped) + ". [{0} chars dropped]".format(chars_dropped) in events[-1]["Message"])
             f.close()
 
-    def test_write_event_msg_size_limit_char_more_than_1_bytes(self):
-        """ Perform 1 byte truncation on char that is more than 1 byte, use decode('utf-8', errors='replace') to replace bad unicode with a good 1 byte char (�) """
+    def test_write_event_msg_size_limit_char_with_actual_unicode_str(self):
+        """ Perform 1 byte truncation on unicode str that is more than 1 byte, use decode('utf-8', errors='replace') to replace bad unicode with a good 1 byte char (�) """
+        message = u"a€bc"*3074  # €(\xe2\x82\xac) is 3 bytes char can be written in windows console w/o encoding
+        self.__assert_write_event_msg_size_limit_char(message)
```

Comment:
Why is it saying all of this isn't covered by tests? Can you investigate this and all the other flagging here?